### PR TITLE
[ecosystem] Create nft nyc flow.

### PR DESCRIPTION
**Pull request by zacharydenton**
## Motivation

Still waiting on UI design but this creates the basic framework which we can add image generation & minting to.

https://user-images.githubusercontent.com/310773/172477951-fad7c044-52c5-4239-8758-6e9042bfa3e3.mp4

## Test Plan

Manual testing

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/aptos-labs/aptos-core/tree/main/developer-docs-site, and link to your PR here.)

### CODE REVIEW

File: ecosystem/platform/server/app/controllers/nft_offers_controller.rb [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  def show; end
+
+  def update
+    nft = Nft.find_by(user: current_user, nft_offer: @nft_offer)
+
+    # TODO: Compute image_url depending on nft_offer.name.
+    image_url = "https://example.com/#{Random.uuid}"
+    nft ||= Nft.create(user: current_user, nft_offer: @nft_offer, image_url:)
+
+    # TODO: Mint the NFT on devnet and save the address.
```
Comment (CapCap) [inline ecosystem/platform/server/app/controllers/nft_offers_controller.rb:20]: One thing here, is that the mint is fully in the FE; 
[Archive.zip](https://github.com/aptos-labs/aptos-core/files/8856608/Archive.zip)

Comment (zacharydenton) [reply ecosystem/platform/server/app/controllers/nft_offers_controller.rb:20]: That should be fine, we can add it to the nft#show page.

File: ecosystem/platform/server/app/controllers/application_controller.rb [hunk 2]
```diff
@@ -42,6 +42,10 @@ def ensure_discord!
     redirect_to root_path unless current_user && current_user.authorizations.where(provider: :discord).exists?
   end
 
+  def ensure_google!
+    redirect_to root_path unless current_user && current_user.authorizations.where(provider: :google).exists?
```
[the remark this answers is not in the document]
Comment (zacharydenton) [reply ecosystem/platform/server/app/controllers/application_controller.rb]: Done.

**Review by CapCap — APPROVED** [review]
/land

**Pull request closed without merging**